### PR TITLE
Implement Unit Testing.

### PR DESCRIPTION
**Pull request by k2b6s9j**
[Review on Reviewable](https://reviewable.io/reviews/atomlinter/linter-clang/110)

### CODE REVIEW

**Comment by k2b6s9j** [comment]
Am I missing something obvious @steelbrain?

**Comment by steelbrain** [comment]
@k2b6s9j investigating

**Comment by k2b6s9j** [comment]
Thank you @steelbrain.

File: spec/linter-clang-spec.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  beforeEach(() => {
+    waitsForPromise(() => {
+      atom.packages.activatePackage("linter-clang");
+    });
+  });
+
+  describe("finds issue with the code", () => {
+    it("in 'missing_import.c'", () => {
+      waitsForPromise(() => {
+        atom.workspace.open("./files/missing_import.c").then((editor) => {
```
Comment (steelbrain) [inline spec/linter-clang-spec.js]: You need to return here, or drop the `{}` of the arrow function. Same for others with promises

Comment (steelbrain) [inline spec/linter-clang-spec.js]: `waitsForPromise` needs a promise returned and it's currently getting `undefined`

**Comment by k2b6s9j** [comment]
I was about to start writing commit messages from Edgar Allen Poe's the Raven because it wasn't working...

File: spec/linter-clang-spec.js (hunk 1 above)
Comment (k2b6s9j) [inline spec/linter-clang-spec.js]: Of course that's it... :frowning: Thanks!

**Comment by k2b6s9j** [comment]
@steelbrain Bad things are still afoot. And I've begun written Poe commit messages...

**Comment by steelbrain** [comment]
@k2b6s9j Hahaha. Are you on slack?

**Pull request merged**